File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int count = 0;
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(numbers, 0, target, 0);
        return count;
    }
    
    public void dfs(int[] numbers, int depth, int target, int result){
        if(depth == numbers.length){
            if(target == result){
            count++;
        }
        return;
    }
    int plus = result + numbers[depth];
    int minus = result - numbers[depth];
    
    dfs(numbers, depth+1, target, plus);
    dfs(numbers, depth+1, target, minus);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    n = len(numbers)
    
    def dfs(idx, result):
        nonlocal answer
        if idx == n:
            if result == target:
                answer+=1
            return
        else:
            dfs(idx+1, result+numbers[idx])
            dfs(idx+1, result-numbers[idx])
    dfs(0,0)
    return answer